text/src/autogluon/text/text_prediction/predictor/predictor.py

```diff
@@ -0,0 +1,382 @@
+        Higher levels correspond to more detailed print statements (you can set verbosity = 0 to suppress warnings).
+        If using logging, you can alternatively control amount of information printed via `logger.setLevel(L)`,
+        where `L` ranges from 0 to 50 (Note: higher values of `L` correspond to fewer print statements, opposite of verbosity levels)
+    warn_if_exist
+        Whether to raise warning if the path exists
+    """
+
+    def __init__(
+            self,
+            label,
+            problem_type=None,
+            eval_metric=None,
+            path=None,
+            verbosity=2,
+            warn_if_exist=True
+    ):
+        self.verbosity = verbosity
+        set_logger_verbosity(self.verbosity, logger=logger)
+        self._label = label
+        self._problem_type = problem_type
+        self._eval_metric = eval_metric
+        self._path = setup_outputdir(path, warn_if_exist=warn_if_exist)
+        self._model = None
+        self._fit_called = False
+        self._backend = None
+
+    def set_verbosity(self, target_verbosity):
+        self.verbosity = target_verbosity
+        set_logger_verbosity(self.verbosity, logger=logger)
+
+    @property
+    def results(self):
+        if self._model is not None:
+            return self._model.results
+        else:
+            return None
+
+    @property
+    def path(self):
+        return self._path
+
+    @property
+    def label(self):
+        return self._label
+
+    @property
+    def problem_type(self):
+        return self._problem_type
+
+    @property
+    def backend(self):
+        return self._backend
+
+    def fit(self,
+            train_data,
+            tuning_data=None,
+            time_limit=None,
+            presets=None,
+            hyperparameters=None,
+            feature_columns=None,
+            column_types=None,
+            num_cpus=None,
+            num_gpus=None,
+            num_trials=None,
+            plot_results=None,
+            seed=None):
+        """Fit the predictor
+
+        Parameters
+        ----------
+        train_data
+            The training data
+        tuning_data
+            The tuning data
+        time_limit
+            The time limits
+        presets
+            The user can specify the presets of the hyper-parameters.
+        hyperparameters
+            The hyper-parameters
+        feature_columns
+            Specify which columns in the data
+        column_types
+            The provided type of the columns
+        num_cpus
+            The number of CPUs to use for each trial
+        num_gpus
+            The number of GPUs to use for each trial
+        num_trials
+            The number of trials. By default, we will use the provided number of trials in the
+            hyperparameters or presets. This will overwrite the provided value.
+        plot_results
+            Whether to plot results.
+        seed
```

Comment:
I think that usually the default value for random_state / seed will be None. (But specifying the seed is also reasonable).

Reply:
I'd prefer setting it to 0 (Tabular's models are all fixed seed by default), this will make benchmarking more straightforward and it will be easier to tell if we've made a change which impacts the model training, plus if we share a script users will get the same result in subsequent runs.

Reply:
Revised.